src/components/current-query/index.js

```diff
@@ -127,67 +141,72 @@ class TextFacetType extends React.Component {
-      text: TextFacetType,
-    };
-
-    return (
-      <LiveMessenger>
-        {({ announcePolite }) => (
-          <React.Fragment>
-            {fields.length > 0 && // Only render this if there are filters applied.
-              <div className="fs-applied-filters">
-                <h2 className="fs-element-invisible">
-                  Currently Applied Search Filters.
-                </h2>
-                <p className="fs-element-invisible">
-                  Click a filter to remove it from your search query.
-                </p>
-                {/* Only render the values for visible facets / filters */}
-                {fields.filter(searchField => !searchField.isHidden).map((searchField, i) => {
-                  // Determine which child component to render.
-                  const MyFacetType = facetTypes[searchField.type];
-                  return (
-                    <MyFacetType
-                      {...this.props}
-                      key={i}
-                      searchField={searchField}
-                      announcePolite={announcePolite}
-                    />
-                  );
-                })}
-              </div>
-            }
-          </React.Fragment>
-        )}
-      </LiveMessenger>
-    );
-  }
-}
+  return (
+    <FacetType onClick={() => removeTextValue(searchField.field)}>
+      {searchField.value}
+    </FacetType>
+  );
+};
+TextFacetType.propTypes = {
+  announcePolite: PropTypes.func.isRequired,
+  onChange: PropTypes.func.isRequired,
+  searchField: PropTypes.shape({
+    field: PropTypes.string.isRequired,
+    value: PropTypes.string.isRequired,
+  }).isRequired,
+};
 
+const FederatedCurrentQuery = (props) => {
+  const { onChange, query } = props;
+
+  const fields = query.searchFields.filter((searchField) => searchField.value
+    && searchField.value.length > 0);
+
+  // Create a map of known facet type child components which can be rendered dynamically.
+  const facetTypes = {
+    'list-facet': ListFacetType,
+    'range-facet': RangeFacetType,
+    text: TextFacetType,
+  };
+
+  return (
+    <LiveMessenger>
+      {({ announcePolite }) => (
+        <>
+          {fields.length > 0 // Only render this if there are filters applied.
+            && (
+            <div className="fs-applied-filters">
+              <h2 className="fs-element-invisible">
+                Currently Applied Search Filters.
+              </h2>
+              <p className="fs-element-invisible">
+                Click a filter to remove it from your search query.
+              </p>
+              {/* Only render the values for visible facets / filters */}
+              {fields.filter((searchField) => !searchField.isHidden).map((searchField) => {
+                // Determine which child component to render.
+                const MyFacetType = facetTypes[searchField.type];
+                return (
+                  <MyFacetType
+                    key={searchField.field}
```

Comment:
Here's another instance of changing the key, so we don't use a loop index value.